### ISSUE TITLE
Code Indexing/Ask Codebase

### ISSUE THREAD

**Issue by Shybella** [ai-genie/chatgpt-vscode#23]
### Describe the feature

**What Is code Indexing / Ask Codebase?**

Code indexing is a technique used to analyze and parse the codebase in order to create an index that allows for faster searching and navigation through code files. This indexing process enhances the speed and ease of finding specific functions or variables in large codebases.

The "ask codebase" feature available on some Visual Studio Code (VSCode) extensions, such as TabNine and Kite, uses machine learning algorithms to provide context-aware coding suggestions. The feature suggests code snippets, function signatures, variable names and other contextual information based on the code being written. By highlighting the code snippet and providing it to GPT, or supplying a compressed/indexed codebase, you can create a list of references of the codebase such as methods and functions that improves coding efficiency and organization.

tl;dr Genie does this now via highlighting the code snippet and supplying that to GPT, the idea is the same but supply a compressed/index codebase instead and create a list of references of the codebase such as methods/functions.

**Comment by genieai-info** [comment]
Thanks for the suggestion. We are planning to enable the indexing feature in the upcoming releases. This is still in planning phase and we will update once we have a release out.

**Comment by Shybella** [comment]
> Thanks for the suggestion. We are planning to enable the indexing feature in the upcoming releases. This is still in planning phase and we will update once we have a release out.

The suggestion was made by GPT-4. 

Thank you.
